models/repo_redirect.go

```diff
@@ -47,7 +50,10 @@ func NewRepoRedirect(ownerID, repoID int64, oldRepoName, newRepoName string) err
 		LowerName:      oldRepoName,
 		RedirectRepoID: repoID,
 	}); err != nil {
-		sess.Rollback()
+		err = sess.Rollback()
+		if err != nil {
```

Comment:
This if statement is not necessary because no matter the outcome err will be returned